spark/client/base/base.go

```diff
@@ -48,5 +48,6 @@ type SparkConnectClient interface {
 
 type ExecuteResponseStream interface {
 	ToTable() (*types.StructType, arrow.Table, error)
+	ToRecordBatches(ctx context.Context) (<-chan arrow.Record, <-chan error, *types.StructType)
```

Comment:
Added so we can consume Apache Arrow record batches rather than the entire table at once, in truth, ToTable should re-use private methods w/ ToRecordBatches, but I've avoided this to avoid breaking clients for now.

Reply:
Thanks! I'll apply this comment (and all the others) in the next round of commits 🥳